src/fsharp/FSharp.Core/array.fs

```diff
@@ -494,12 +494,23 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Filter")>]
         let filter f (array: _[]) = 
-            checkNonNull "array" array
-            let res = List<_>() // ResizeArray
```

Comment:
@ReedCopsey.  Allocated bytes is not really a big deal.  # of Allocations is kind of the issue.  pre-allocating the list will eliminate additional allocations due to growing the list, presumably reducing allocations to 2.